libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -49,6 +49,13 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 			if (!dynamic_cast<FunctionDefinition const*>(declaration))
 				return declaration;
 	}
+	else if (dynamic_cast<EventDefinition const*>(&_declaration))
+	{
+		// check that all other declarations with the same name are events
```

Comment:
That means at this point of the code, we need to see the arguments not only the names.  I'll try.